pdp/contract/utils_test.go

```diff
@@ -0,0 +1,62 @@
+package contract
+
+import (
+	"bytes"
+	"testing"
+
+	"github.com/ethereum/go-ethereum/common"
+)
+
+// TestDecodeAddressCapability tests the canonical address decoding.
+func TestDecodeAddressCapability(t *testing.T) {
+	// Case 1: Input shorter than 20 bytes → left pad with zeros
+	input1 := []byte{0x01, 0x02, 0x03}
+	// Use HexToAddress for string constants
+	expected1 := common.HexToAddress("0x0000000000000000000000000000000000010203")
+	decoded1 := DecodeAddressCapability(input1)
+	if !bytes.Equal(decoded1.Bytes(), expected1.Bytes()) {
+		t.Errorf("Case 1 (short) failed: Expected %s, got %s", expected1.Hex(), decoded1.Hex())
+	}
+
+	// Case 2: Exactly 20 bytes input → use as is
```

Comment:
just make these into separate test cases instead of cramming them into the same test case